.github/workflows/build-publish-develop-pr.yml

```diff
@@ -0,0 +1,76 @@
+name: "Build and Publish Chainlink"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+
+jobs:
+  goreleaser-build-publish-chainlink:
+    runs-on:
+      labels: ubuntu22.04-16cores-64GB
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+
+      - name: Get image tag
+        id: get-image-tag
+        run: |
+          short_sha=$(git rev-parse --short HEAD)
+          if [[ ${{ github.event_name }} == 'push' ]]; then
+            echo "image-tag=develop-${short_sha}" >> $GITHUB_OUTPUT
+          else
+            if [[ ${{ github.head_ref }} == 'release/'* ]]; then
+              echo "image-tag=release-${short_sha}" >> $GITHUB_OUTPUT
+            else
+              echo "image-tag=pr-${{ github.event.number }}-${short_sha}" >> $GITHUB_OUTPUT
+            fi
+          fi
+
+      - name: Configure aws credentials
+        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+        with:
+          role-to-assume: ${{ secrets.AWS_OIDC_IAM_ROLE_BUILD_PUBLISH_DEVELOP_PR }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          mask-aws-account-id: true
+          role-session-name: goreleaser-build-publish-chainlink
+
+      - name: Build and publish images
+        id: goreleaser-build-publish
+        uses: ./.github/actions/goreleaser-build-sign-publish
+        with:
+          enable-docker-publish: "true"
```

Comment:
Do we think it's useful to have all PR's publish or only those with a certain label? I worry about the hard ECR limits (10k images) and the volume of PR's + commits on those PR's.

Reply:
ahh yeah good call. I guess only pr + label `build` will trigger this and if the pr with branch name `release/*` it will auto build as well as push to develop?

Reply:
Yeah, if we push to release/* or trunk (develop) branches, we should publish. But maybe we can filter PR builds to only publish if there's a label? In case the label is added after the fact, we'd have to add job triggering from label events too.